mitodl/mit-open
# Replace react-dotdotdot with CSS

### What are the relevant tickets?
Relates to
- https://github.com/mitodl/hq/issues/4205 particularly comment https://github.com/mitodl/hq/issues/4205#issuecomment-2105413877
- https://github.com/mitodl/course-search-utils/issues/75

### Description (What does it do?)
This PR replaces `react-dotdotdot` with CSS. For the search page, the motivation is described in https://github.com/mitodl/course-search-utils/issues/75. Elsewhere... `react-dotdotdot` seems not to work with our line heights and `neue-haas-grotesk`. I do not entirely understand why. That said, this css-based solution has good browser support and other advantages. E.g., it doesn't literally truncate the text string, just the displayed value. That seems desirable for screenreaders, particularly in the case of search page facets.

### Screenshots (if appropriate):

[Storybook](https://shy-guide.surge.sh/?path=/docs/ol-components-truncatetext--docs)

<img width="870" alt="Screenshot 2024-05-10 at 7 55 18 PM" src="https://github.com/mitodl/mit-open/assets/9010790/0400e311-046c-4110-8c1c-48c84eae5f51">

<img width="250" alt="Screenshot 2024-05-10 at 7 55 52 PM" src="https://github.com/mitodl/mit-open/assets/9010790/b5ddcac0-3710-499e-890d-bedd38bc3639">

### How can this be tested?
1. Reinstall js dependencies. If you're running the app in docker, you'll want to `docker compose build watch`.
2. View http://localhost:6006/?path=/docs/ol-components-truncatetext--docs in several browsers
3. View the search page. Long facets should be truncated to one line. You might need to narrow your browser window.
